### DISCUSSION TITLE
Is there an easy way to exclude the current element from selectors?

### DISCUSSION THREAD

**Discussion by midgleyc** [jhy/jsoup#2251] [Q&A] [answered]
As mentioned in the docs, selectors may match the current element or any of its children. Is there a simple (or recommended) way to only match children? At the moment I'm filtering out the current element if it was matched
```java
node.getElementsByTag("table").stream().filter(x -> x != node).toList();
```

**Comment by jhy** [comment, marked as answer]
Hi,

There are a few ways to do this. The cleanest syntax might be to select on the element's `children()`:

```
String html = "<div id=1>Outer <div id=2>One</div><div id=3>Two</div></div>";
Document doc = Jsoup.parse(html);
Element el = doc.expectFirst("div");
Elements inner = el.children().select("div");
```

Or, modify the selector to only match the children directly:

```
Elements inner = doc.select("div div");
```

**Reply by midgleyc** [reply]
Thanks, I don't necessarily know what the higher node is (in the code, only that I don't want to select it), but `children().select` is perfect.

**Reply by jhy** [reply]
On review - a simpler and more efficient way is just to write the query as a descendant of the context element:

```java
String html = "<div id=1>Outer <div id=2>One</div><div id=3>Two</div></div>";
Document doc = Jsoup.parse(html);
Element el = doc.expectFirst("div");
Elements inner = el.select("* div");
```

In jsoup we support starting the CSS query with a combinator, which will be applied to the root (context) element. Using that vs `children()` will be somewhat more efficient as the query can execute in one pass. Plus it looks cleaner, IMHO.

Or use `> div` to find the direct descendents.
